### PR TITLE
Homework 6

### PR DESCRIPTION
**Pull request by AghajanyanA**

### CODE REVIEW

File: ui/src/components/homePage/homePage/auth.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public loginTitle = this.LOCATORS.loginTitle
+
+    public registerTitle = this.LOCATORS.registerTitle
+
+
+    public async displayRegisterPage(): Promise<void> {
+        await this.LOCATORS.createNewAccountBTN.click()
+        await expect(this.LOCATORS.registerTitle).toBeVisible()
```
Comment (vladimirchizhenkov) [inline ui/src/components/homePage/homePage/auth.ts:20]: We don't use expects in tests

Comment (AghajanyanA) [reply ui/src/components/homePage/homePage/auth.ts]: Fixed

File: ui/src/components/homePage/homePage/auth.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    protected LOCATORS = {
+        loginTitle: this.locator.locator('//h2[contains(text(), "Access")]'),
+        createNewAccountBTN: this.locator.locator('//span[text()="Create UHCGlasses.com Account"]'),
+        registerTitle: this.locator.locator('//h2[text()="No vision insurance? We got you!"]'),
+        welcomeWindow: this.locator.locator('//div[@class="rc-dialog-content"]'),
+    }
+
+    public loginTitle = this.LOCATORS.loginTitle
+
+    public registerTitle = this.LOCATORS.registerTitle
```
Comment (vladimirchizhenkov) [inline ui/src/components/homePage/homePage/auth.ts:14]: Don't use properties as getters of LOCATORS

File: ui/src/components/homePage/homePage/eligiblityWidget.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        header: this.locator.locator('//header[contains(@class, "eligibilityWidget__header")]/p')
+    }
+
+    public window(): Locator {
+        return this.locator
+    }
+
+    public header(): Locator {
+        return this.LOCATORS.header
+    }
```
Comment (vladimirchizhenkov) [inline ui/src/components/homePage/homePage/eligiblityWidget.ts:15]: Don't use methods as getters for LOCATORS, please

File: ui/src/components/homePage/homePage/header/auth/register.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    email: string, 
+    password: string
+}
+
+export class Register extends Component {
+    private LOCATORS = {
+        firstName: this.locator.locator('//input[@name="firstName"]'),
+        lastName: this.locator.locator('//input[@name="lastName"]'),
+        email: this.locator.locator('//input[@name="email"]'),
+        password: this.locator.locator('//input[@name="password"]'),
```
Comment (vladimirchizhenkov) [inline ui/src/components/homePage/homePage/header/auth/register.ts:15]: You can use this fields like this:
private LOCATORS = {
input: (name: string Or any type) => this.locator.locator(`//input[@name="${name}"]`)

File: ui/src/components/homePage/homePage/header/auth/register.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        await this.LOCATORS.password.fill(data.password)
+    }
+
+    public async createNewAccount(data: RegistrationForm) {
+        await this.fillRegisterData(data)
+        await this.LOCATORS.createAccountBTN.click()
+    }
+
+    
+
```
Comment (vladimirchizhenkov) [inline ui/src/components/homePage/homePage/header/auth/register.ts:31]: Use  eslint && prettier, please
yarn eslint --fix

File: ui/src/components/homePage/homePage/header/topSide.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+    public welcomeMessage(): Locator {
+        return this.LOCATORS.welcomeMessage
+    }
+
+    public myAccount(): Locator {
+        return this.LOCATORS.myAccount
+    }
+
+    public MyAccountDrop = new MyAccountDrop(this, this.LOCATORS.myAccountContainer, this.page)
```
Comment (vladimirchizhenkov) [inline ui/src/components/homePage/homePage/header/topSide.ts:24]: This does not work. The component only takes 2 arguments.

File: ui/tests/UHC-1-POM.test.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        })    
+        
+        await test.step('Step 4 — Check if user is successfully registered', async () => {       
+            await homePage.Welcome.closeWindow();
+
+            await expect(homePage.Welcome.welcomeWindow()).toBeHidden();
+
+            await expect(homePage.Header.TopSide.welcomeMessage()).toHaveText(`Hello, ${userName}`);
+
+            const capitalizedUsername = userName.charAt(0).toUpperCase() + userName.slice(1)
```
Comment (vladimirchizhenkov) [inline ui/tests/UHC-1-POM.test.ts:53]: Move the implementation of this function to the POM component

File: ui/src/components/homePage/homePage/eligiblityWidget.ts [hunk 8]
```diff
@@ -1,16 +1,16 @@
...
 
-    public window(): Locator {
-        return this.locator
+    public async isVisible(): Promise<boolean> {
+        return await this.locator.isVisible();
     }
 
-    public header(): Locator {
-        return this.LOCATORS.header
+    public async headerText(): Promise<string | null> {
```
Comment (vladimirchizhenkov) [inline ui/src/components/homePage/homePage/eligiblityWidget.ts:13]: it is a function -> **get**HeaderText

**Pull request merged**